src/steadystateproblem.cpp

```diff
@@ -209,6 +203,69 @@ void SteadystateProblem::findSteadyStateBySimulation(Solver *solver,
     }
 }
 
+bool SteadystateProblem::initializeBackwardProblem(Solver *solver,
+                                                   Model *model,
+                                                   const BackwardProblem *bwd) {
+    if (bwd) {
+        /* If preequilibration but not adjoint mode, there's nothing to do */
+        if (solver->getSensitivityMethodPreequilibration() !=
+            SensitivityMethod::adjoint)
+            return false;
+
+        /* If we need to reinitialize solver states, this won't work yet. */
+        if (model->nx_reinit() > 0)
+            throw NewtonFailure(AMICI_NOT_IMPLEMENTED,
+                "Adjoint preequilibration with reinitialization of "
+                "non-constant states is not yet implemented. Stopping.");
+
+        /* If we have a backward problem, we're in preequilibration.
+           Hence, quantities like t, x, and xB must be set. */
+        solver->reInit(t, x, x);
+        solver->updateAndReinitStatesAndSensitivities(model);
+        xB.copy(bwd->getAdjointState());
+    }
+
+    /* Will need to write quadratures: resize */
+    xQB.reset();
```

Comment:
this doesn't resize, it just sets all entries to zero.

Reply:
Was an resize originally, just didn't update the comment. Thanks! ;)